source/common/config/well_known_names.cc

```diff
@@ -135,8 +135,8 @@ TagNameValues::TagNameValues() {
   // http.(<stat_prefix>.)*
   addTokenized(HTTP_CONN_MANAGER_PREFIX, "http.$.**");
 
-  // listener.(<address>.)*
-  addRe2(LISTENER_ADDRESS, R"(^listener\.((<ADDRESS>)\.))");
+  // listener.(<address|stat_prefix>.)*, but specifically excluding "admin"
+  addRe2(LISTENER_ADDRESS, R"(^listener\.((<ADDRESS>|<TAG_VALUE>)\.))", "", "admin");
```

Comment:
Note that in `std::regex` this could all be part of the regex, as this engine supports negative matches. But I don't want to add any more uses of std::regex if it's all possible to avoid, due to performance issues.

Reply:
I agree with your call here.